editor/src/clj/editor/gui.clj

```diff
@@ -3934,6 +3936,42 @@
         (protobuf/assign-repeated :resources merged-resource-descs)
         (update :material #(or % default-material-proj-path)))))
 
+(defn- add-dropped-resource
+  [selection workspace resource]
+  (let [scene (node->gui-scene (first selection))
+        ext (str/lower-case (resource/ext resource))
+        base-name (resource/base-name resource)
+        gen-name #(->> (g/node-value (g/node-value scene %) :name-counts)
+                       (outline/resolve-id base-name))]
+    (cond
+      (= ext "particlefx")
+      (add-particlefx-resource scene (g/node-value scene :particlefx-resources-node) resource (gen-name :particlefx-resources-node))
+
+      (= ext "font")
+      (add-font scene (g/node-value scene :fonts-node) resource (gen-name :fonts-node))
+
+      (some #{ext} (workspace/resource-kind-extensions workspace :atlas))
+      (add-texture scene (g/node-value scene :textures-node) resource (gen-name :textures-node))
```

Comment:
This case should include `["atlas" "tilesource"]` and custom formats if I understand this correctly.